src/ethereum/forks/amsterdam/vm/__init__.py

```diff
@@ -142,7 +143,8 @@ class Message:
     accessed_storage_keys: Set[Tuple[Address, Bytes32]]
     disable_precompiles: bool
     parent_evm: Optional["Evm"]
-    transaction_state_changes: StateChanges
+    is_create: bool = False
```

Comment:
I think in case of `Message` and `TransactionEnvironment`, I'd prefer not to rely on default values of the `dataclass`. I think the specs are more readable when we explicitly define all the attributes of these objects at the time of instantiation. Same for `BlockEnvironment`